pkg/operations/kubernetesupgrade/upgradeagentnode.go

```diff
@@ -87,11 +88,23 @@ func (kan *UpgradeAgentNode) CreateNode(poolName string, agentNo int) error {
 		kan.ParametersMap,
 		nil)
 
-	if err != nil {
-		return err
+	if err == nil {
+		return nil
 	}
 
-	return nil
+	kan.logger.Errorf("Deployment %s failed with error %v", deploymentName, err)
+	// Get deployment error details
+	errRespList, e := armhelpers.GetDeploymentError(kan.Client, kan.logger, kan.ResourceGroup, deploymentName)
+	if e != nil || len(errRespList) == 0 {
+		kan.logger.Errorf("Failed to get error details for deployment %s: %v", deploymentName, e)
+		// return original deployment error
+		return err
+	}
+	errStrList := make([]string, len(errRespList))
+	for i, errResp := range errRespList {
```

Comment:
I thought you are changing it to return an array. why changed your mind?

Reply:
I returned an array, but eventually have to turn it into an error